lib/open_food_network/payments_report.rb

```diff
@@ -100,7 +100,7 @@ def columns
          proc { |orders| orders.first.distributor.name },
          proc { |orders| orders.to_a.sum(&:item_total) },
          proc { |orders| orders.sum(&:ship_total) },
-         proc { |orders| orders.sum(&:outstanding_balance) },
+         proc { |orders| orders.sum { |order| order.outstanding_balance.to_f } },
```

Comment:
I wonder if you could still use the old syntax if you implemented `+` on OutstandingBalance. Maybe it should be a delegator.

Reply:
Nice one! I didn't think how `#sum` works under the hood but what you say is very likely. I can check it's source code 👌

Reply:
this is sweet @mkllnk . It even feels good :joy: 

```
Failures:                                                                                                                                                                                                                                                         1) Payments Reports shows orders with payment state, their balance and totals                                                                                                                                                                                      Failure/Error: proc { |orders| orders.sum { |order| order.outstanding_balance } },                                                                                                              NoMethodError:                                                                                                             
       undefined method `+' for #<OrderBalance:0x0000557dbf8c5ea0>                                                              
     # /bundles/gems/activesupport-5.0.7.2/lib/active_support/core_ext/enumerable.rb:37:in `inject'                             
     # /bundles/gems/activesupport-5.0.7.2/lib/active_support/core_ext/enumerable.rb:37:in `sum'                                
     # /bundles/gems/activesupport-5.0.7.2/lib/active_support/core_ext/enumerable.rb:143:in `sum'                               
     # /bundles/gems/activesupport-5.0.7.2/lib/active_support/core_ext/enumerable.rb:35:in `sum'                                
     # /bundles/gems/activesupport-5.0.7.2/lib/active_support/core_ext/enumerable.rb:143:in `sum'                               
     # ./lib/open_food_network/payments_report.rb:103:in `block in columns'                                                     
     # ./lib/open_food_network/order_grouper.rb:67:in `build_cell'                                                              
     # ./lib/open_food_network/order_grouper.rb:73:in `block in build_row'
```

thanks for a lot for this feedback!

Reply:
See https://github.com/openfoodfoundation/openfoodnetwork/pull/7150/commits/65b569fe5891c6109912f21f50fd7b0e9bbad270